library/src/com/squareup/timessquare/CalendarCellView.java

```diff
@@ -42,28 +42,38 @@ public CalendarCellView(Context context, AttributeSet attrs) {
   }
 
   public void setSelectable(boolean isSelectable) {
-    this.isSelectable = isSelectable;
-    refreshDrawableState();
+    if (this.isSelectable != isSelectable) {
+      this.isSelectable = isSelectable;
+      refreshDrawableState();
+    }
   }
 
   public void setCurrentMonth(boolean isCurrentMonth) {
-    this.isCurrentMonth = isCurrentMonth;
-    refreshDrawableState();
+    if (this.isCurrentMonth != isCurrentMonth) {
+      this.isCurrentMonth = isCurrentMonth;
+      refreshDrawableState();
+    }
   }
 
   public void setToday(boolean isToday) {
-    this.isToday = isToday;
-    refreshDrawableState();
+    if (this.isToday != isToday) {
+      this.isToday = isToday;
+      refreshDrawableState();
+    }
   }
 
   public void setRangeState(MonthCellDescriptor.RangeState rangeState) {
-    this.rangeState = rangeState;
-    refreshDrawableState();
+    if (this.rangeState != rangeState) {
+      this.rangeState = rangeState;
+      refreshDrawableState();
+    }
   }
 
   public void setHighlighted(boolean highlighted) {
-    isHighlighted = highlighted;
-    refreshDrawableState();
+    if (this.isHighlighted != isHighlighted) {
```

Comment:
This line should be `if (this.isHighlighted != highlighted) {`. The parameter is named `highlighted` and not `isHighlighted`.

Alternatively the parameter could be renamed to `isHighlighted` to match the semantics of the other setters.